### ISSUE TITLE
Problem with custom transition

### ISSUE THREAD

**Issue by ugogo** [fkhadra/react-toastify#85]
Hey,

I'm trying to update the default animation, but the `enter` animation does not seem to be triggered.
I'd like to avoid to manually add and remove classes on the `node`, so I tried to do something with the `Transition` component.

```javascript
// css

const animationWrapperCSS = css({
  transition: `transform ${ANIMATION_DURATION}ms ease-out`,
  transform: 'translateY(100%)',

  '&.is-entering': {
    transform: 'translateY(0)',
  },

  '&.is-entered': {
    transform: 'translateY(0)',
  },

  '&.is-exiting': {
    transform: 'translateY(100%)',
  },

  '&.is-exited': {
    transform: 'translateY(100%)',
  },
});
```

```javascript
// custom transition component

const AnimationWrapper = ({ children, ...props }) => (
  <Transition
    {...props}
    timeout={ANIMATION_DURATION}
  >
    {state => (
      <div className={`is-${state} ${animationWrapperCSS}`}>
        { children }
      </div>
    )}
  </Transition>
);
```

```javascript
// container

<ToastContainer
  autoClose={false}
  transition={AnimationWrapper}
  position={toast.POSITION.BOTTOM_CENTER}
/>
```

I created a repo if you want to have a look: https://github.com/ugogo/toastify-debug

Thanks!

**Comment by fkhadra** [comment]
Hey @ugogo,

You are trying to do something like that  guess ? (sorry for the big gif btw)
![toastify-custom-transition](https://user-images.githubusercontent.com/5574267/33425930-7f55075e-d5c0-11e7-9302-ecfa33423295.gif)

tl;dr
In your code, you are using the 4 states provided by the Transition. You could simplify it by using only 2 of them:

```js
//I use 150% instead of 100% so the element goes out of the screen
const animationWrapperCSS = css({
  transition: `transform ${ANIMATION_DURATION}ms ease-out`,
 // will be the initial frame, so we can skip entering state
  transform: 'translateY(150%)', 
    '&.is-entered': {
    transform: 'translateY(0)',
  },
  // remove exiting cause it's a kind of initial frame before exit
  '&.is-exited': {
    transform: 'translateY(150%)',
  },
});
```
What happened to your code:

The transition pass the state in the following order:

- trigger the toast
1. exited
2. entering
3. entered 
- close the toast
4. entered
5. exiting
6. exited 

- `entering` is the initial frame. In your case the value of entering refer to the final frame
- I'll suggest using all the states only when you need it(complex animation). For that kind of transition is just fine to stick with 2 states. I did the same for the default one.

I hope my explanation can help you.

**Comment by ugogo** [comment]
Hah, so it was only a matter of state.
Thanks for the explanation and the fast answer 🤘
